[add examples-by-tas]

hiowenluke opened the pull request:
Rewrote the examples with Tas. The code structure is clearer, which helps user better understand Express and Node.js and master modular development in Express and Node.js. 

![compare-with-tas-mvc-index js](https://raw.githubusercontent.com/tasjs/express-examples-by-tas/master/res/mvc-index.png)

![compare-with-tas-mvc-lib-boot js](https://raw.githubusercontent.com/tasjs/express-examples-by-tas/master/res/mvc-lib.png)

hiowenluke committed:
[add examples-by-tas]

Rewrote the examples with Tas. The code structure is clearer, which helps user better understand Express and Node.js and master modular development in Express and Node.js. See details in READM.md.

wesleytodd commented:
Hey @hiowenluke!  I would be interested to know what you intend for this PR.  My guess is that it is not in the interest of the team to have two example sets, one which is heavily reliant on a specific third-party module.  It would seem to me this was more of an exercise to show the possibilities of using your library, correct?  Could you elaborate on why this is opened as a PR or what you hoped to achieve?

dougwilson commented:
At the very least, it looks like when you cloned the existing tests, it wasn't even the current version of the tests at the time of your PR. It looks to have been at least a month or more out of date. For example, we removed all the Jade usage from the examples since that is an abandoned project.

hiowenluke committed:
remove comments of install tas

hiowenluke committed:
use the newest resource

hiowenluke committed:
use ejs engine

hiowenluke committed:
use ejs engine

hiowenluke committed:
add guideline

hiowenluke committed:
add the redis installation guide

hiowenluke committed:
simplified readme

hiowenluke committed:
fix examples/view-constructor

http://localhost:3000/
ReferenceError: https is not defined

hiowenluke committed:
fix examples/route-separation

http://localhost:3000
Click link "posts", jumped to the "users" page.

hiowenluke committed:
fix examples/params

http://localhost:3000/users/0-2
users tj, tobi     <-- lost loki

Because user/0 is tj, that’s means the 0 is the data, not index of
array.
So, users/0-2 must be include user/2.

hiowenluke committed:
fix examples/markdown

http://localhost:3000/fail
Error: Failed to lookup view "missing" in views directory

hiowenluke committed:
fix examples/online

// first (if you have not done it yes):
// $ npm install redis
// $ brew install redis   <-- lost this line
// $ redis-server

If the user has installed redis, then there is no problem reinstalling
redis. But if here does not prompt the user to install redis, this
example will run failure, the user will be confused.

hiowenluke committed:
fix examples/search

// first (if you have not done it yes):
// $ npm install redis
// $ brew install redis   <-- lost this line
// $ redis-server

If the user has installed redis, then there is no problem reinstalling
redis. But if here does not prompt the user to install redis, this
example will run failure, the user will be confused.

hiowenluke committed:
fix examples/session

// first (if you have not done it yes):
// $ npm install redis
// $ brew install redis   <-- lost this line
// $ redis-server

If the user has installed redis, then there is no problem reinstalling
redis. But if here does not prompt the user to install redis, this
example will run failure, the user will be confused.

hiowenluke committed:
cancel fixing examples/session

hiowenluke committed:
cancel fixing examples/search

hiowenluke committed:
cancel fixing examples/online

hiowenluke committed:
Revert "cancel fixing examples/online"

This reverts commit 8e9d27872846c3f607bf270e736b77d34f251f60.

hiowenluke committed:
Revert "fix examples/view-constructor"

This reverts commit 93278b463d8c1d7dc68c8a2389bc226ea15d9334.

hiowenluke committed:
Revert "fix examples/route-separation"

This reverts commit 0cd90d931c1d83c551435a22161b3ac66b62afa2.

hiowenluke committed:
Revert "fix examples/params"

This reverts commit 5238255a969c067690b0eef6e61a1c551d0acfad.

hiowenluke committed:
Revert "fix examples/markdown"

This reverts commit e7d67772726d3fffc6affbd7a489d539487da2e5.

hiowenluke committed:
Revert "fix examples/online"

This reverts commit 9e8970b24e85292f1d1182869c9e7b5995d55fa4.

hiowenluke committed:
Revert "cancel fixing examples/session"

This reverts commit a140026f068500b42c00241f0031e62e2edbdcdb.

hiowenluke committed:
Revert "cancel fixing examples/search"

This reverts commit 2f94b84d8f5bd9c0adc0a006d380862639ec59e2.

hiowenluke committed:
Revert "cancel fixing examples/online"

This reverts commit 8e9d27872846c3f607bf270e736b77d34f251f60.

dougwilson commented:
Hi @hiowenluke I'm not sure what is going on in this PR, as it seems to have evolved into way more than just "Rewrote the examples with Tas." as it looks like there are other fixes, other changes, etc. It's very difficult to understand what is going on. Can you please break apart the PR into multiple to better understand what the changes going on in here are?

The changes seem to highlight the fears brought up by @wesleytodd in that since it adds an entire second set of example, they seem to be drifting apart in functionality, with bug fixes and new functionality getting added in the tas examples, but not the originals.
